lua/ltex_extra/src/log.lua

```diff
@@ -1,42 +1,51 @@
-local debug = false
-
 local log = {}
 
-if debug then
-    log = require("plenary.log").new {
-        plugin = "ltex_extra",
-        use_file = false,
-        level = "trace"
-    }
-else
-    local dummy = function (...) return true end
-    log.trace     = dummy
-    log.debug     = dummy
-    log.info      = dummy
-    log.warn      = dummy
-    log.error     = dummy
-    log.fatal     = dummy
-    log.fmt_trace = dummy
-    log.fmt_debug = dummy
-    log.fmt_info  = dummy
-    log.fmt_warn  = dummy
-    log.fmt_error = dummy
-    log.fmt_fatal = dummy
-end
+log.init = function (level)
+    if type(level) == "string" and level ~= "none" then
+        if not pcall(require, "plenary.log") then
+            vim.notify("`ltex_extra` requires the `plenary` plugin to display logs", vim.log.levels.WARN)
+            return
+        end
+        local tmp_log = require("plenary.log").new {
+            plugin = "ltex_extra",
+            use_file = false,
+            level = level
+        }
+        for k, v in pairs(tmp_log) do
```

Comment:
I don't know why you are doing this.. Plenary logs new return a kind of logger instance, not an iterable..